src/Microsoft.AspNet.Mvc.Razor/RazorView.cs

```diff
@@ -233,13 +238,18 @@ private Task RenderPageCoreAsync(IRazorPage page, ViewContext context)
             }
         }
 
-        private IRazorPage GetLayoutPage(ViewContext context, string layoutPath)
+        private IRazorPage GetLayoutPage(ViewContext context, string executingFilePath, string layoutPath)
         {
-            var layoutPageResult = _viewEngine.FindPage(context, layoutPath);
+            var layoutPageResult = _viewEngine.GetPage(executingFilePath, layoutPath, isPartial: true);
```

Comment:
The parameter name lines up with the `IsPartial` properties in `IRazorPage` and `RazorView`. I have not changed what pages / views are considered partial, just exposed the possibility of creating a non-partial page using `GetPage()` and `FindPage()`.

FYI only top-level views i.e. those `ViewResultExecutor` processes are _not_ considered partial.